services/alerts/src/useAlert.ts

```diff
@@ -1,27 +1,51 @@
-import { useContext } from 'react'
+import { useContext, useRef, useCallback, useMemo } from 'react'
 import { AlertsManagerContext } from './AlertsManagerContext'
-import { Alert, AlertOptions } from './types'
+import {
+    AlertBarComponent,
+    AlertOptions,
+    AlertsManager,
+    AlertsManagerAlert,
+} from './types'
 
 export const useAlert = (
     message: string | ((props: any) => string),
     options: AlertOptions | ((props: any) => AlertOptions) = {}
 ) => {
-    const alertsManager = useContext(AlertsManagerContext)
+    const alertsManager: AlertsManager = useContext(AlertsManagerContext)
+    const ref = useRef<AlertBarComponent>()
+    const alert: AlertsManagerAlert = useMemo(() => alertsManager.add(ref), [
+        alertsManager,
+    ])
 
-    const show = (props?: any) => {
-        const resolvedMessage = String(
-            typeof message === 'function' ? message(props) : message
-        )
-        const resolvedOptions =
-            typeof options === 'function' ? options(props) : options
+    const show = useCallback(
+        (props?) => {
+            const resolvedMessage = String(
+                typeof message === 'function' ? message(props) : message
+            )
+            const resolvedOptions =
+                typeof options === 'function' ? options(props) : options
 
-        const alert: Alert = {
-            message: resolvedMessage,
-            options: resolvedOptions,
-        }
+            alertsManager.show({
+                ...alert,
+                message: resolvedMessage,
+                options: resolvedOptions,
+            })
+        },
```

Comment:
Should this function return `() => alert.hide()` so that it can be used in a closure?

Reply:
Could you show a (pseudo code) example of why this would be useful, and offers benefits over using the `hide` function returned from the hook?

I'm not sure what the use would be, but I'm probably not considering the type of use-case you had in mind.